proto/tendermint/abci/types.proto

```diff
@@ -260,6 +266,11 @@ message ResponseApplySnapshotChunk {
   }
 }
 
+message ResponsePreprocessTxs {
+  repeated bytes            txs      = 1;
+  tendermint.types.Messages messages = 2;
+}
```

Comment:
In a follow-ip issue, this will also return/include the intermediate state roots. Then, in another PR, this will enable immediate execution (ref #3 and https://github.com/tendermint/spec/issues/208#issuecomment-721158558)